### PR TITLE
Add more default dirs to serve and compile

### PR DESCRIPTION
**Pull request by jakemac53**
... but only actually serve them if they exist.

Fixes https://github.com/dart-lang/build/issues/1006
Fixes https://github.com/dart-lang/build/issues/1007

### CODE REVIEW

File: build_web_compilers/build.yaml [hunk 1]
```diff
@@ -11,20 +11,31 @@ builders:
         - .ddc.js.map
     is_optional: True
     auto_apply: all_packages
-    required_inputs: [".dart"]
+    required_inputs:
+      - ".dart"
```
Comment (natebosch) [inline build_web_compilers/build.yaml]: [nit] inconsistent quoting with line 27

Out of curiosity - is there a reason to prefer this style over the more compact version?

File: build_web_compilers/build.yaml [hunk 2]
```diff
@@ -11,20 +11,31 @@ builders:
...
   entrypoint:
     target: "build_web_compilers"
     import: "package:build_web_compilers/builders.dart"
-    builder_factories: ["webEntrypointBuilder"]
+    builder_factories:
+      - "webEntrypointBuilder"
```
Comment (natebosch) [inline build_web_compilers/build.yaml]: [nit] inconsistent quoting

File: build_web_compilers/build.yaml [hunk 3]
```diff
@@ -11,20 +11,31 @@ builders:
...
     build_to: cache
     auto_apply: root_package
     defaults:
       generate_for:
-        include: ["web/**", "test/**_test.dart"]
-        exclude: ["test/**.node_test.dart", "test/**.vm_test.dart"]
+        include:
+          - web/**
+          - test/**_test.dart
+          - example/**
```
Comment (natebosch) [inline build_web_compilers/build.yaml]: I'm a bit concerned about this change. It means we'll spam errors when there are non-web entrypoints in example/

File: build_web_compilers/build.yaml (hunk 1 above)
Comment (jakemac53) [reply build_web_compilers/build.yaml]: Just for consistency, we have other lists that would end up wrapping. We could change everything to this style too, but it feels a bit weird in yaml to me.

File: build_web_compilers/build.yaml (hunk 3 above)
Comment (jakemac53) [reply build_web_compilers/build.yaml]: Only if there are non-web examples in example, and you have a dependency on build_web_compilers. The idea is just to try and do the right most of the time - I think its fairly safe to assume web entrypoints everywhere if somebody has this dependency.

**Pull request merged**